### ISSUE TITLE
Breaking change in 2.1.0

### ISSUE THREAD

**Issue by HunderlineK** [markedjs/marked#2106]
**Marked version: 2.1.0**

**Describe the bug**
The latest code uses the optional chaining operator which makes marked 2.1.0 incompatible with Node V12. It's breaking semantic release which uses Marked. Here:
https://github.com/markedjs/marked/commit/5be9d6d70ea35be6398ee35958fb8ac955a89fbe

Thanks!

**Comment by calculuschild** [comment]
@UziTech Shouldn't Babel be able to handle this?

**Comment by glasser** [comment]
This is affecting our Node 12 CI as well.

**Comment by UziTech** [comment]
`const marked = require("marked")` in node uses [/src/marked.js](https://github.com/markedjs/marked/blob/master/src/marked.js) not the transpiled es5 version at [/lib/marked.js](https://github.com/markedjs/marked/blob/master/lib/marked.js)

The problem is that marked is only tested in `lts/*`(v14 currently) and `latest` (v16 currently) but the `?.` syntax doesn't work in v12.

We had a discussion in https://github.com/markedjs/marked/pull/1938#discussion_r574175702 about what versions of node marked should support

**Comment by glasser** [comment]
(Dropping support for Node 12 is totally reasonable — just publish a rollback version at 2.1.1 and publish the version that doesn't support Node 12 as 3.0.0.)

**Comment by utsavkapoor** [comment]
Broke our CI too, via pulling in protobufjs - upgrading to node 14+ seems get past the issue, but we can't do that as a solution for other reasons.

**Comment by glasser** [comment]
Yes, I'm discovering that the protobufjs CLI runs a bunch of unversioned (!) npm installs at runtime (!) which pull in marked recursively.

**Comment by utsavkapoor** [comment]
@glasser - Has anyone reported this to protobufjs ?

**Comment by glasser** [comment]
protobufjs is barely maintained. we use our own fork of it and I'm trying to figure out a way to untangle its web of hackery, but honestly if marked@2.1.1 supported Node 12 and marked@3.0.0 was what you published as 2.1.0 it would solve this problem.  "No longer runs under a major version of Node that's still in LTS" is a pretty clearly semver-major change.

**Comment by calculuschild** [comment]
> const marked = require("marked") in node uses /src/marked.js not the transpiled es5 version at /lib/marked.js

@UziTech Should Marked supply the es5 version by default? The transpiled version seems to run slightly faster on the benchmarks anyway. We already have Babel doing the work for us but our documentation doesn't give any hint on how to access it for people who need that compatibility.

**Comment by UziTech** [comment]
> The transpiled version seems to run slightly faster on the benchmarks anyway.

It [didn't](https://github.com/markedjs/marked/pull/1563#issuecomment-550399786) when we started transpiling it which is why we stuck with main pointing to the es6 version, but it seems to have gotten faster over time so maybe we should?

**Comment by glasser** [comment]
(We're working around it by changing our project's build system to not run the protobufjs CLI automatically on every install but just manually with checking in the generated code, which means the CLI doesn't have to run on every Node version we support.)

**Comment by brendanmc6** [comment]
For those running into this problem with Next.js on Vercel:

```
22:00:37.460 | Error occurred prerendering page "/404". Read more: https://nextjs.org/docs/messages/prerender-error
22:00:37.460 | /vercel/path0/node_modules/marked/src/marked.js:158
22:00:37.460 | const prevRenderer = extensions.renderers?.[ext.name];
```

You can fix this by upgrading the node version to 14x in your Vercel project settings!

**Comment by david-amores-anz** [comment]
This is also breaking [Twilio CLI's flex plugin](https://github.com/twilio/flex-plugin-builder/tree/main/packages/flex-dev-utils):

```
$ DEBUG=* twilio flex:plugins:deploy --patch --changelog="test deploy"

/Users/xxxxxx/.twilio-cli/node_modules/marked/src/marked.js:158
          const prevRenderer = extensions.renderers?.[ext.name];
                                                    ^

SyntaxError: Unexpected token '.'
    at compileFunction (<anonymous>)
    at wrapSafe (internal/modules/cjs/loader.js:1053:16)
    at Module._compile (internal/modules/cjs/loader.js:1101:27)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1157:10)
    at Module.load (internal/modules/cjs/loader.js:985:32)
    at Function.Module._load (internal/modules/cjs/loader.js:878:14)
    at Module.require (internal/modules/cjs/loader.js:1025:19)
    at require (internal/modules/cjs/helpers.js:72:18)
    at Object.<anonymous> (/Users/xxxxxx/.twilio-cli/node_modules/flex-dev-utils/dist/marked.js:7:32)
    at Module._compile (internal/modules/cjs/loader.js:1137:30)
```

**Comment by iamwwc** [comment]
tldr have broken on node 12 https://tldr.sh/ 😭

**Comment by 4eb0da** [comment]
I've found some workaround, maybe it will help someone

Call it before using protobufjs / pbjs
```
cd node_modules/protobufjs/cli && npm i && npm i marked@2.0
```

**Comment by josefschabasser** [comment]
Hi! This issue just bit us.


> I've found some workaround, maybe it will help someone
> 
> Call it before using protobufjs / pbjs
> 
> ```
> cd node_modules/protobufjs/cli && npm i && npm i marked@2.0
> ```

Is this really the only solution right now? We're using yarn v2 and tried the `resolutions` field in our `package.json` but it doesn't work. `yarn why marked` yields no output at all.

**Comment by Nilos** [comment]
We're using yarn too and fixed this in the following way:
```
yarn add --dev marked@2.0.3
yarn add --dev jsdoc@^3.6.3
```
Alternatively you can also force marked@^2.0.3 to use 2.0.3.

The main thing is to install jsdoc yourself.

**Comment by HunderlineK** [comment]
As a workaround for our use case, we fixed the semantic-release  to `"semantic-release": "17.3.6",`

**Comment by taycaldwell** [comment]
> 🎉 This issue has been resolved in version 2.1.1 🎉
> 
> The release is available on:
> 
> * [npm package (@latest dist-tag)](https://www.npmjs.com/package/marked/v/2.1.1)
> * [GitHub release](https://github.com/markedjs/marked/releases/tag/v2.1.1)
> 
> Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot 📦🚀

This issue still exists in version 2.1.1.

Revert and release breaking change in a new major version please.

**Comment by josefschabasser** [comment]
> > tada This issue has been resolved in version 2.1.1 tada
> > The release is available on:
> > 
> > * [npm package (@latest dist-tag)](https://www.npmjs.com/package/marked/v/2.1.1)
> > * [GitHub release](https://github.com/markedjs/marked/releases/tag/v2.1.1)
> > 
> > Your **[semantic-release](https://github.com/semantic-release/semantic-release)** bot packagerocket
> 
> This issue still exists in version 2.1.1.
> 
> Revert and release breaking change in a new major version please.

For us (local and CI) it's resolved. Maybe you should delete your `node_modules` and reinstall?
